app/uk/gov/hmrc/helptosavefrontend/models/EmailVerificationRequest.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.models
+
+import play.api.libs.json._
+
+case class EmailVerificationRequest(email: String, templateId: String,
+                                    linkExpiryDuration: String, continueUrl: String,
+                                    templateParameters: Map[String, String])
+
+object EmailVerificationRequest {
+  implicit val formats: Format[EmailVerificationRequest] = Json.format[EmailVerificationRequest]
+
+  def apply(email: String, nino: String, templateId: String, linkExpiryDuration: String, continueUrl: String,
+            templateParameters: Map[String, String]): EmailVerificationRequest =
+    EmailVerificationRequest(email, templateId, linkExpiryDuration, continueUrl,
+      Map("email" → email, "nino" → nino) ++ templateParameters)
```

Comment:
for now this is how we are checking the user that logged in, is the user being redirected via the email